### DISCUSSION TITLE
Issue with NodePort on MacOS (Ventura 13.0.1)

### DISCUSSION THREAD

**Discussion by evakkuri** [strimzi/strimzi-kafka-operator#7653] [Q&A] [answered]
Hi! This is a pretty general question, so I did not know which repository to use. If this is not a proper channel, please direct me to the correct place.

I'm trying to follow the quickstart documentation from here: https://strimzi.io/docs/operators/latest/quickstart.html. I have installed minikube and set up the cluster according to the instructions. However, when I'm trying to produce messages to the cluster using the minikube node IP, I only get connection errors.

What I have found out so far is that the node IP is not where minikube actually responds, it responds simply via `localhost`. If I try to connect to the external bootstrap say via `localhost:32072`, the first connection succeeds, and the client finds the IP address of the broker to which to connect. However, the broker IP found is the non-functioning minikube node IP, so the following connections fail. I can access the actual broker just fine with `localhost:{node port of broker}`, but same issue with the subsequent connections.

The exact same issue presents itself if I try to surface the service with `minikube service ... --url`, and with Docker Desktop's own Kubernetes.

Any ideas how to correct this?

**Comment by scholzj** [comment, marked as answer]
Hi @evakkuri 

This is absolutely the best place to ask ;-).

However, the answer here is not easy, because things are complicated here. `nodeport` type listener exposes the service on a port on the Kubernetes worker node. However, on MacOS, there is no native support for Docker containers. So when you have Docker running on MacOS, it essentially runs a VM with Linux inside where it runs the containers. And then, when you run Minikube with the default `docker` driver, it is basically started only as a Docker container inside the VM for running the Docker containers. So out of the box, the node port is exposed on the Docker container, but not on the VM.

There are two general solutions:
* Use your Minikube to use a different driver and run in its own VM. That way, the node port should be exposed directly on the VM and should be accessible.
* Try to find out what is the _right_ hostname under which the port is accessible and configure it as _advertised_. You can do that in the `Kafka` custom resource: https://strimzi.io/docs/operators/latest/full/configuring.html#type-GenericKafkaListenerConfigurationBroker-reference ... and that you should be able to just connect to it from your Kafka client. From what you say, it sounds like `localhost` might work for you? So setting that should make it work also for the follow-up connections

(Unfrotunately, these things depedn heavily on all the different versions and configurations used - so it is really hard to write a _definitive_ guide how should it work and give some better answers than _try this or that_ :-()

**Reply by evakkuri** [reply]
Thank you @scholzj , setting the `advertisedHost` in the broker settings worked like a charm :)
